src/main/webapp/app/quiz/edit/multiple-choice-question/edit-multiple-choice-question.component.ts

```diff
@@ -97,114 +71,98 @@ export class EditMultipleChoiceQuestionComponent implements OnInit, OnChanges, A
-        this.questionEditor.getEditor().clearSelection();
-        const lines = this.questionEditorText.split('\n').length;
-        const range = this.questionEditor.getEditor().selection.getRange();
-        this.questionEditor.getEditor().moveCursorTo(this.questionEditor.getEditor().getCursorPosition().row, Number.POSITIVE_INFINITY);
-        this.questionEditor.getEditor().insert(textToAdd);
-        range.setStart(lines, 4);
-        range.setEnd(lines, textToAdd.length - 1);
-        this.questionEditor.getEditor().selection.setRange(range, false);
+    prepareForSave(): void {
+        this.cleanupQuestion();
+        this.markdownEditor.parse();
+        this.questionUpdated.emit();
     }
 
     /**
-     * @function addHintAtCursor
-     * @desc Adds the markdown for a hint at the current cursor location
+     * @function cleanupQuestion
+     * @desc Clear the question to avoid double assignments of one attribute
      */
-    addHintAtCursor(): void {
-        this.artemisMarkdown.addHintAtCursor(this.questionEditor.getEditor());
+    private cleanupQuestion() {
+        // Reset Question Object
+        this.question.answerOptions = [];
+        this.question.text = null;
+        this.question.explanation = null;
+        this.question.hint = null;
+        this.question.hasCorrectOption = null;
     }
 
     /**
-     * This function opens the modal for the help dialog.
+     * @function domainCommandsFound
+     * @desc Get the {array} from the editor and assign its values based on the domainCommands to the corresponding question attributes one by one
+     *       1. Go trough each element of the {array} domain command
+     *       2. If the first element {string} has not second element {domainCommandIdentifier} in the array assign the string to the question text
+     *       3. else assign the first element {string} according to the second element {domainCommandIdentifier} to the corresponding attributes of the question
+     * @param {array} of markdownTextLine {string} with the corresponding domainCommand {DomainCommand} identifier
      */
-    open(content: any) {
-        this.modalService.open(content, { size: 'lg' });
-    }
+    domainCommandsFound(domainCommands: [string, DomainCommand][]): void {
+        this.cleanupQuestion();
+        let currentAnswerOption = new AnswerOption();
 
-    /**
-     * @function addExplanationAtCursor
-     * @desc Adds the markdown for an explanation at the current cursor location
-     */
-    addExplanationAtCursor(): void {
-        this.artemisMarkdown.addExplanationAtCursor(this.questionEditor.getEditor());
+        for (const command of domainCommands)  {
+
+            if (command[1] === null && command[0].length > 0) {
+                this.question.text = command[0];
+            }
+            if (command[1] instanceof CorrectOptionCommand || command[1] instanceof IncorrectOptionCommand) {
+                currentAnswerOption = new AnswerOption();
```

Comment:
Why is this initialized again here (see at the top of the function)?

Reply:
Unchanged.

Reply:
After our chat I now understand what is happening in this function.
Please update the JsDoc as discussed.

Reply:
The documentation is still not satisfying to me. I'll provide more information why:

    /**
     * @function domainCommandsFound
     * @desc 1. Gets the {array} containing the text with the domainCommandIdentifier and creates a new multiple choice question
_=> The first part is not precise enough, it is a tuple of text and domainCommandIdentifiers
=> The last part seems wrong to me as no multiple choice question is created, rather attributes are assigned to this component, as you mention later in the JsDoc_

     *       by assigning the text according to the domainCommandIdentifiers to the multiple choice question attributes.
_=> ok_
     *       (question text, explanation, hint, answeroption(correct/wrong)
_=> In my opinion this is the most important information that needs to be provided, but it seems out of context here. There should be a small explanation stating that the order of the array entries is relevant and follows the structure that you describe here. However it is misleading, because as far as I see the structure can be repeating (e.g. text, explanation, answeroption, text, answeroption, but not answeroption, answeroption or hint, explanation._

     *       2. For each answer option a new AnswerOption entity is created by assiging the answeroption text and setting
     *       the answeroption correct/wrong
=> I don't understand this sentence
     *       3.Important is that the resetMultipleChoicePreview() is triggered to notify the parent component
_=> I would remove "Important is that" and just state that the mentioned method is triggered._
     *       about the changes within the question and to trigger the checking method for correct values
_=> This sentence is hard to understand - what is the checking method for correct values?_
Why does this need to happen when the method domainCommandsFound is triggered?
     * @param {array} of markdownText {string} with the corresponding domainCommand {DomainCommand} identifier
_=> array is not the name of the parameter, that is domainCommands. Also it is not an array with x and y, but an array of tuples [x, y]._ 
     */